### ISSUE TITLE
Cannot access a property or method of a null object reference. Star.swf

### ISSUE THREAD

**Issue by matthias-hoste** [lightspark/lightspark#1070]
## In order to help us understand your issue, please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your issue (like this: `[x]`)
- Please fill in the sections relevant to your issue and delete the rest.
- Use the *Preview* tab to see what your issue will actually look like

---

### What is the purpose of your issue?
- [ ] Site Support Request: Lightspark does not work correctly on a specific website.
- [x] Bug: Lightspark has the same problem across multiple websites or SWF files.
- [ ] Feature request: a request for new features or support for new platforms, *not sites or SWF files*.
- [ ] Question, Comment, or Suggestion
- [ ] If none of the above apply, check this box.

---

### The following sections should be filled in based on the purpose listed above. Erase any section (the contents between triple ---) that does not apply.

---

#### Provide the complete URL of the page with the issue here:

https://game.galaxylifegame.net/game

#### If it is possible to download the file and run it with the command line, please provide a link to the file here:

[URL of file](https://cdn.galaxylifegame.net/Star.swf)

#### Provide the complete set of instructions required to reproduce the problem:

1. Start using command line
2. Ignore unhandled exceptions(there is a skippable one)
3. Spam of nullreference

#### What kind of failure is your issue?

- [ ] **Fault**: You receive a black window instead of the flash content with a white text such as ``We're sorry, Lightspark encountered an as yet unsupported Flash file``
- [ ] **Crash**: The window disappears or your browser crashes with a message such as ``A tab has crashed``
- [x] **Broken**: Lightspark shows something, but most of the file is not usable
- [ ] **Glitch**: Lightspark shows something and most of the file is usable, but things are missing or wrong compared to official Flash.
- [ ] **Performance**: Lightspark runs slow or uses lots of memory trying to play the file

#### Provide the Lightspark version

Output of ``lightspark -v``:

```
INFO: Lightspark version 0.8.6-git Copyright 2009-2013 Alessandro Pignotti and others
```

#### What operating system are you are running on?

- [x] Windows 7, 8 or 10
- [ ] Windows XP, Vista or older
- [ ] Linux: *<specify distro and version here, such as Ubuntu 18.04>*
- [ ] Mac OS X: *<specify version here, such as 10.8.1>*
- [ ] Other: *<specify OS and version here>*

#### How are you are running Lightspark?

- [ ] Browser plugin for Chrome or Chromium: *<specify Chromium version here>*
- [ ] Browser plugin for Waterfox: *<specify Waterfox version here>*
- [ ] Browser plugin for other browser: *<specify browser and version here>*
- [x] Lightspark executable on the command line

#### Which of these actions you have taken to help debug the problem?

- [ ] I attached a screen shot of the problem to this issue
- [ ] I have a backtrace listing and pasted it in the bottom section of this issue (*hint: hitting Ctrl-C after a crash will copy it to the clipboard*)
- [ ] I have reproduced the problem using the Lightspark command line executable
- [x] I ran `lightspark -l 4` on the command line and am attaching logs to this issue (*hint: if they are too big, zip them*)

```ERROR: Unhandled ActionScript exception in VM Error: CheatManager is a singleton class, use getInstance() instead
INFO: dynamic cast:Error
ERROR: Unhandled ActionScript exception in VM Error: CheatManager is a singleton class, use getInstance() instead
    at com.reygazu.anticheat.managers::CheatManager/()
    at Class/getInstance()
    at com.reygazu.anticheat.variables::SecureObject/()
    at Class/()
    at global/()
    at com.dchoc.game.core.flow::Application/loadDoDoStep()
    at com.dchoc.game.core.flow::Application/loadDoStep()
    at com.dchoc.game.core.flow::Application/load()
    at com.dchoc.game.core.flow::Application/logicUpdate()
    at com.dchoc.game.core.flow::Application/logicUpdate()
    at com.dchoc.game.core.flow::Application/onEnterFrame()

CALLS: INIT_FRAME
CALLS: EXECUTE_FRAMESCRIPT
INFO: FPS: 45 0
CALLS: ADVANCE_FRAME
CALLS: Handling event enterFrame 0x3dbd250
INFO: throwing exception:0x3dbd250 id:1009 Error #1009: Cannot access a property or method of a null object reference.
ERROR: Unhandled ActionScript exception in VM TypeError: Error #1009: Error #1009: Cannot access a property or method of a null object reference.
INFO: dynamic cast:TypeError
ERROR: Unhandled ActionScript exception in VM TypeError: Error #1009: Error #1009: Cannot access a property or method of a null object reference.
    at com.dchoc.game.core.resource::ResourceMng/buildDoSyncStep()
    at com.dchoc.game.core.resource::ResourceMng/build()
    at com.dchoc.game.core.flow::Application/childrenBuild()
    at com.dchoc.game.core.flow::Application/build()
    at com.dchoc.game.core.flow::Application/logicUpdate()
    at com.dchoc.game.core.flow::Application/logicUpdate()
    at com.dchoc.game.core.flow::Application/onEnterFrame()
    at com.dchoc.game.core.flow::Application/onEnterFrame()
```

### Description of your issue

*Explanation of your issue in arbitrary form goes here.*

**Comment by dbluelle** [comment]
I've fixed that error, but now there is another exception thrown...
The exception seems to be related to the flash vars that are missing when starting the swf file from the standalone application.
I tried running it in a webbrowser, but unfortunately I wasn't able to create an account on that website (didn't get an email after registration), so I can't test it...
If you create a file containing the flash vars (by setting the LIGHTSPARK_PLUGIN_PARAMFILE environment variable) and upload that here, that may help.

**Comment by matthias-hoste** [comment]
Amazing! I had created the file before luckily :D if there's anything wrong with it let me know and I will change it
[stage.txt](https://github.com/lightspark/lightspark/files/11524161/stage.txt)

**Comment by dbluelle** [comment]
Yes, this error is because of the security settings.
I'm currently using the following command line:
`lightspark -u https://lb.galaxylifeserver.net/star/game -s local-with-networking -p stage.txt Star.swf`
That works and starts downloading stuff but then throws another exception...

**Comment by matthias-hoste** [comment]
Indeed, after I applied an extra check ensuring the exception does not trigger it only gives a black screen after downloading the resources queue xml. I'll check to see what it could be stuck on but I assume it's a bit further

**Comment by matthias-hoste** [comment]
So I got this:

```Kibble — Today at 10:30 AM
it's trying to download a bunch of stuff from https://cdn.galaxylifegame.net when it should be downloading them from https://cdn.galaxylifegame.net/content
I'm running lightspark -u https://cdn.galaxylifegame.net/content -s local-with-networking -p stage.txt Star.swf tho
oh I just need to do /content/ instead of /content```

It then crashes on flash.utils.getQualifiedClassName which is fixed by adding --avmplus
It then starts actually loading into the game but at a very slow pace of 1fps and some things being offset
https://cdn.discordapp.com/attachments/917397552058482688/1110854999359094824/image.png

**Comment by matthias-hoste** [comment]
I hear you thinking "ah shit this guy again".
Yep :)

I tried out the latest changes which progress quite a lot further but once it starts loading the images it goes wrong.
Log is here and above are the commands I'm still using:

https://pastebin.com/Kyjq7can

**Comment by dbluelle** [comment]
It seems that the swf is working, but at some point during the loading screen it wants to connect to a server by the ExternalInterface which isn't available in the standalone application. So it never proceeds past that point.
Unfortunately I'm still unable to login to the site (even resetting the password doesn't help, The website claims to have sent an e-mail but it doesn't appear in my mailbox). 
When trying to register again it says that my email address is already registered.
So I can't test if it works when using the browser plugin...

**Comment by matthias-hoste** [comment]
> It seems that the swf is working, but at some point during the loading screen it wants to connect to a server by the ExternalInterface which isn't available in the standalone application. So it never proceeds past that point.
> 
> Unfortunately I'm still unable to login to the site (even resetting the password doesn't help, The website claims to have sent an e-mail but it doesn't appear in my mailbox). 
> 
> When trying to register again it says that my email address is already registered.
> 
> So I can't test if lightspark works when using the browser plugin...
> 
> 

That is annoying indeed, I'll give it a try. As for the website, if you send an email to administration@phoenixnetwork.net from the registered email you can get the forgot password email. Sorry that it doesnt work :( seems to have issues with some mail providers. Great thanks for the help though!

**Comment by matthias-hoste** [comment]
Alright so I tried it :D it appears to do all of the initial loading but then freezes, even going so far as to not let lightspark respond to anything anymore(have to kill with task manager)
Right before this happens it printed:
```(lightspark.exe:58824): Pango-WARNING **: All font fallbacks failed!!!!
_cairo_win32_scaled_font_set_metrics:GetTextMetrics: The operation completed successfully.

(lightspark.exe:58824): Pango-CRITICAL **: pango_font_description_set_size: assertion 'size >= 0' failed

(lightspark.exe:58824): Pango-WARNING **: failed to create cairo scaled font, expect ugly output. the offending font is 'Times New Roman, Bold'

(lightspark.exe:58824): Pango-WARNING **: font_face status is: no error has occurred

(lightspark.exe:58824): Pango-WARNING **: scaled_font status is: error occurred in the Windows Graphics Device Interface```

And then during that it spams this over and over:

```NOT_IMPLEMENTED: getDefaultTextFormat does not get all fields of TextFormat
NOT_IMPLEMENTED: setDefaultTextFormat does not set all fields of TextFormat```
